nilearn/_utils/niimg.py

```diff
@@ -108,40 +114,41 @@ def load_niimg(niimg, dtype=None):
 
     Returns
     -------
-    img : image
+    img : spatialimages.SpatialImage
         A loaded image object.
     """
-    from ..image import new_img_like  # avoid circular imports
+    from nilearn.image import new_img_like  # avoid circular imports
 
-    niimg = stringify_path(niimg)
-    if isinstance(niimg, str):
+    tmp = stringify_path(niimg)
+    if isinstance(tmp, (str)):
         # data is a filename, we load it
-        niimg = load(niimg)
-    elif not isinstance(niimg, spatialimages.SpatialImage):
+        loaded_niimg = load(tmp)
+    elif isinstance(tmp, spatialimages.SpatialImage):
+        loaded_niimg = tmp
+    else:
         raise TypeError(
             "Data given cannot be loaded because it is"
             " not compatible with nibabel format:\n"
-            + repr_niimgs(niimg, shorten=True)
+            + repr_niimgs(tmp, shorten=True)
         )
 
-    dtype = _get_target_dtype(_get_data(niimg).dtype, dtype)
+    assert isinstance(loaded_niimg, spatialimages.SpatialImage)
 
-    if dtype is not None:
-        # Copyheader and set dtype in header if header exists
-        if niimg.header is not None:
-            niimg = new_img_like(
-                niimg,
-                _get_data(niimg).astype(dtype),
-                niimg.affine,
-                copy_header=True,
-            )
-            niimg.header.set_data_dtype(dtype)
-        else:
-            niimg = new_img_like(
-                niimg, _get_data(niimg).astype(dtype), niimg.affine
-            )
```

Comment:
I don't think that is ever used to load non spatial images so they always have a header